src/python/gem5/resources/client_api/jsonclient.py

```diff
@@ -54,21 +55,47 @@ def __init__(self, path: str):
         self.path = path
         self.resources = []
 
-        if Path(self.path).is_file():
-            self.resources = json.load(open(self.path))
-        elif not self._url_validator(self.path):
-            raise Exception(
-                f"Resources location '{self.path}' is not a valid path or URL."
-            )
-        else:
-            req = request.Request(self.path)
+        # Try loading as local file if it exists
+        if Path(path).is_file():
             try:
-                response = request.urlopen(req)
+                with open(path, encoding="utf-8") as f:
+                    self.resources = json.load(f)
+                return
+            except json.JSONDecodeError as e:
+                raise ValueError(f"Invalid JSON in file '{path}': {e}")
+            except Exception as e:
+                raise FileNotFoundError(f"Error reading file '{path}': {e}")
+
+        # Handle URLs (including file:// URIs)
+        if self._url_validator(path):
+            try:
+                parsed_url = urllib.parse.urlparse(path)
+                # Handle file:// URLs
+                if parsed_url.scheme == "file":
+                    local_path = Path(parsed_url.path)
+                    if not local_path.is_file():
+                        raise FileNotFoundError(f"File not found: '{path}'")
+
+                    with local_path.open("r", encoding="utf-8") as f:
+                        self.resources = json.load(f)
+                    return
+
+                # Handle HTTP/HTTPS URLs
+                req = request.Request(path)
+                with request.urlopen(req) as response:
+                    self.resources = json.loads(
+                        response.read().decode("utf-8")
+                    )
```

Comment:
This may be something you want to punt to another PR, but is this going to break for people who are using a proxy?

Reply:
This will probably fail for people using proxy with an actual url as a JSON. This would still fail currently (as this part of the code is essentially the same). I will update it in another PR and add proxy context to the `urlopen` like it is done in `atlasclient.py` https://github.com/gem5/gem5/blob/c9625ce9cc5b5a90a38327de5ac0e1870974af5e/src/python/gem5/resources/client_api/atlasclient.py#L138

Reply:
Sounds good. Just make sure to minimize the duplicated code when you do that :)